library/src/scala/tasty/reflect/Kernel.scala

```diff
@@ -865,6 +865,14 @@ trait Kernel {
   /** Is this type an instance of a non-bottom subclass of the given class `cls`? */
   def Type_derivesFrom(self: Type)(cls: ClassDefSymbol)(implicit ctx: Context): Boolean
 
+  /** Is this type a function type?
+   *
+   *  @return true if the dealised type of `self` without refinement is `FunctionN[T1, T2, ..., Tn]`
+   *
+   *  @note `List[Int]` is not a function type, despite that `List[Int] <:< Int => Int`.
+   */
+  def Type_isFunctionType(self: Type)(implicit ctx: Context): Boolean
```

Comment:
`Type_isFunctionType` returns true for `given Int => Int` and `erased Int => Int`. The documentation should reflect that.

Given that we are adding this check we should also have `Type_isImplicitFunctionType` and `Type_isErasedFunctionType` to be able to distinguish them.